### ISSUE TITLE
Bar Chart options misleading

### ISSUE THREAD

**Issue by jricoiii** [isle-project/isle-editor#1149]
**Is your feature request related to a problem? Please describe.**
n/a

**Describe the solution you'd like**
When creating a bar chart of a variable group by a second variable, it seems to me the response variable should be the main variable (not the group by variable).  The title is even additionally misleading "variable given by group by variable"

I think it should be the response variable as the Variable and the explanatory as teh Group by

**Describe alternatives you've considered**
N/A

**Additional context**
Here's an example of marital status broken by sex (gender)
![image](https://user-images.githubusercontent.com/50716185/87492357-413bec80-c5ff-11ea-8515-50ffb2b09557.png)

**Comment by gweinberg** [comment]
^^ Agree, it's backwards.  For instance, if I want to make a relative frequency bar graph of smoking (yes/no) "grouped by gender" or "given gender," I'd expect:
![bargraph1](https://user-images.githubusercontent.com/31780210/87614480-1f656700-c6de-11ea-920f-155122e8e3c6.png)
where the primary (response) variable of interest is smoking (yes/no), and it's conditioned on ("given") the explanatory variable gender; so the heights of the bars are P(smoke|M), P(not|M), P(smoke|F), and P(not|F).  Then by the law of total probability, the conditional proportions of smoking (yes/no) sum to 100% within each explanatory (gender) group.
But in ISLE, it reverses the roles of the variable you're selecting and the variable you're "grouping by" or "given," and I can't figure out what the heights of the bars represent (they certainly don't sum to 100% within each cluster on the graph).

**Comment by Planeshifter** [comment]
The barchart options have now been overhauled following your feedback.

You can try out the new behavior in this data explorer:

https://isle.stat.cmu.edu/demonstrations/heartdisease-barchat/

When using relative frequencies, one can choose either to display the overall relative frequencies as they would appear in a contingency table (via the `TotalPercent` option) or the conditional relative frequencies that Gordon Weinberg asked for when choosing `Relative frequencies inside each group`.

Please let me know if the barchart options make sense or whether you would rename / change any behavior.

Thanks!

**Comment by gweinberg** [comment]
Thanks!

1) Suggestion on labels:

![image](https://user-images.githubusercontent.com/31780210/89474611-181d0080-d754-11ea-8211-fd5c7ea859d4.png)

(I think we've discussed before, a suggestion for multi-level labels)

2) As I'm testing it out, I'm finding (to echo other requests that have been made) that I really want a button to "show previous dialog" which would pull up the most recently used toolbox dialog, with all the settings already entered as they had been (Minitab, for instance, has "edit previous dialog"); that would make returning to the same dialog box a lot less time consuming.  [I can make that as a separate request if you want.]

3) The "order bars by frequency" option is still backwards -- I'd think that if I select "order bars", that it would involve ordering the 0/1/2 bars of the inner-most variable (in my example above);  but I see that isn't what it's doing, it's actually re-ordering the groups of the outermost (conditioned-on) variable.  That's backwards; and especially because it says "...by frequency", but "frequency of GENDER" isn't even plotted on the graph now,    When "group by" is activated, I don't really see a reason to have a separate pull down just to re-order the clusters on the plot (something like that could perhaps be better doable by somehow making the plot clickable, like maybe when you double-click on "male" or something, it toggles the ordering of the genders on the plot back and forth.  Anyway, having that pulldown say "reorder bars by frequency" is confusing now.  If you wanted to keep it, then when "group by" is activated, it would need to say something like, "order outer groups by..." (but then, what would you order them by?  Alphabetically?  It's confusing to order them by some measure in the data that I'm not showing on the plot.  I don't even quite get what it's flipping them with respect to; total count of male vs total count of female?  That's confusing, since neither of those things are the heights of any of my bars.

**Comment by Planeshifter** [comment]
Hi Gordon,

thanks for all the thoughtful comments!

Concerning each of the points:

1) Changed the positioning of the x-axis labels to not give the false impression as to refer to only one of the bars when grouping. We cannot easily map multiple variables to the x-axis, so right now not possible to show then also the "0", "1", "2" on the x-axis. Noted as a potential future improvement though.
2) Thanks for catching this bug, which is now fixed.
3) Yes, I think it would be best to create a separate issue for that. For the Fall, we should be able to roll out the history log as demonstrated in the [following JSM poster](https://www.youtube.com/watch?v=cAZOrT3sP7Q), which will allow one to recreate any previous output. Maybe one could add a button for each logged action that would bring up the respective toolbox options, so that one could make adjustments when recreating the output.
4) Have some ideas on how to refactor this to be more generally useful, will upload a new data explorer to play around with here once that work is finished.
5) Agreed. I will work on improving the default titles and labels.

**Comment by Planeshifter** [comment]
Quick update: Concerning bullet point four, the OkCupid explorer at https://isle.stat.cmu.edu/demonstrations/okcupid/
now has new, hopefully better explained, options for re-ordering the x-axis values. Chose OkCupid as an example as the data has many categorical variables which one might want to order by frequency.

**Comment by gweinberg** [comment]
hmm.... I still think saying "order x axis" sounds like it should order *everthing* on the x axis, and then I'm surprised when the orders of the bars don't change within each group.  (E.g., barchart -> count of distinct values -> variable ="diet" group by = "sex"), then selecting "order x axis" sounds like it should re-order the bars within each group (since they're on the x axis, and they're the primary thing I'm looking at).
If "group by" is selected, then I think you don't mean "order x axis", you mean something like "re-order outer groups by..." 

By the way, is mouseover supposed to pull up the descriptions of *all* the bars?  It would be better if it pulled up the description of just the bar being moused-over (or maybe a toggle between functionality).  Also, for some reason, in the order of the popup descriptions, it moved the one fror the first (blue) bar ("null") and it moved the one for the purple ("thin"), and it left gaps in the list from where the descriptions had been moved:

![image](https://user-images.githubusercontent.com/31780210/89681012-17a27800-d8c2-11ea-8104-a04c99890b2a.png)

**Comment by Planeshifter** [comment]
Closing as we now a) display "Order outer groups" when a grouping variable is selected instead of "Order x axis" and b) the mouseover only pulls up the information for the nearest bar.
